### PR TITLE
[Bug]: Failed to Connect to MO using "MySQL Workbench" #4530

### PR DESCRIPTION
**Pull request by badboynt1**
add variable sql_safe_updates
add support show collation statement

## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4530

## What this PR does / why we need it:
add variable sql_safe_updates
add support show collation statement

### CODE REVIEW

**Review by aunjgr — APPROVED** [review]

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y [hunk 1]
```diff
@@ -2107,7 +2107,13 @@ show_stmt:
 |   show_target_filter_stmt
 |   show_table_status_stmt
 |   show_grants_stmt
+|   show_collation_stmt
 
+show_collation_stmt:
+    SHOW COLLATION like_opt where_expression_opt
+    {
+        $$ = &tree.ShowCollation{}
```
Comment (ouyuanning) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.y:2115]: may be you lost your like/ where field?

Comment (ouyuanning) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.y:2115]: and you can add ut in  mysql_sql_test.go

File: pkg/sql/plan/build_show.go [hunk 2]
```diff
@@ -447,6 +447,12 @@ func buildShowStatus(stmt *tree.ShowStatus, ctx CompilerContext) (*Plan, error)
 	return returnByRewriteSQL(ctx, sql, ddlType)
 }
 
+func buildShowCollation(stmt *tree.ShowCollation, ctx CompilerContext) (*Plan, error) {
+	ddlType := plan.DataDefinition_SHOW_COLLATION
+	sql := "select 'utf8mb4_bin' as `Collation`, 'utf8mb4' as `Charset`, 46 as `Id`, 'Yes' as `Compiled`, 1 as `Sortlen`"
+	return returnByRewriteSQL(ctx, sql, ddlType)
```
Comment (ouyuanning) [inline pkg/sql/plan/build_show.go:453]: i don't known if you want to support like/where ?

Comment (badboynt1) [reply pkg/sql/plan/build_show.go:453]: > 

for now only 1 collation is supported, so only 1 row was returned. like/where is only supported in parser now, and dont actually work. will add it later when more collations supported

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y (hunk 1 above)
Comment (badboynt1) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.y:2115]: > and you can add ut in mysql_sql_test.go

fixed

**Review by ouyuanning — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Pull request merged**